app/controllers/user_actions_controller.rb

```diff
@@ -28,7 +28,16 @@ def index
     }
 
     stream = UserAction.stream(opts).to_a
-    render_serialized(stream, UserActionSerializer, root: "user_actions")
+
+    response = { user_actions: serialize_data(stream, UserActionSerializer) }
+
+    if guardian.can_lazy_load_categories?
+      category_ids = stream.map(&:category_id).compact.uniq
+      categories = Category.secured(guardian).with_parents(category_ids)
+      response[:categories] = serialize_data(categories, CategoryBadgeSerializer)
+    end
```

Comment:
A spec for this and the drafts controller would be good as well. Especially to make sure we don't drop this once we fully switch to lazy-loaded categories.